README.md

```diff
@@ -101,35 +101,76 @@ const libp2p = require('libp2p')
 const TCP = require('libp2p-tcp')
 const WS = require('libp2p-websockets')
 const spdy = require('libp2p-spdy')
+const mplex = require('libp2p-mplex')
 const secio = require('libp2p-secio')
 const MulticastDNS = require('libp2p-mdns')
 const DHT = require('libp2p-kad-dht')
 
 class Node extends libp2p {
-  constructor (peerInfo, peerBook, options) {
-    options = options || {}
-
-    const modules = {
-      transport: [
-        new TCP(),
-        new WS()
-      ],
-      connection: {
-        muxer: [
+  constructor (_peerInfo, _peerBook, _options) {
+    const options = {
+      peerInfo: _peerInfo             // The Identity of your Peer
+      peerBook: _peerBook             // Where peers get tracked, if undefined libp2p will create one instance
+      modules: {
+        transport: [
```

Comment:
I think some kind of first and second class transport system is needed.
For example: the new ws-star needs the libp2p swarm to dial the server via WS but WS is independent from the swarm.
So any listen/dial stuff for first class transports would be setup before the second class transports are setup (so dialing does not fail)

Reply:
Both transports will be first class with #130 occurring during the #159. Not the concern of this proposal/PR

Reply:
we had a proposal for transport priorities at some point, but it was rejected.

Reply:
@dryajov link it please

Reply:
@diasdavid https://lodash.com/docs/4.17.5#defaults and https://lodash.com/docs/4.17.5#defaultsDeep

Reply:
<img width="589" alt="image" src="https://user-images.githubusercontent.com/1211152/36448563-ebe3367a-167f-11e8-87d0-caaf136906f3.png">

I'm confused, wrong c&p?

Reply:
my bad, this is the correct one - https://github.com/libp2p/js-libp2p-tcp/pull/78

Reply:
@dryajov that was a proposal for a pseudo priority based on what the module says. It needs to be a user defined.